app/controllers/application_controller/ci_processing.rb

```diff
@@ -978,6 +970,30 @@ def process_elements(elements, klass, task, display_name = nil, order_field = ni
 
   private ############################
 
+  def process_element_destroy(element, klass, name)
+    return unless element.respond_to?(:destroy)
+
+    audit = {:event        => "#{klass.name.downcase}_record_delete",
+             :message      => "[#{name}] Record deleted",
+             :target_id    => element.id,
+             :target_class => klass.base_class.name,
+             :userid       => session[:userid]}
+
+    model_name  = ui_lookup(:model => klass.name) # Lookup friendly model name in dictionary
+    record_name = get_record_display_name(element)
+
+    element.destroy
```

Comment:
Something like the following can trap both `rescue` and `destroyed?` result when calling `destroy` method.

```ruby
begin
  element.destroy
rescue => err
  destroyed = false
  reason = err.message
elsif element.destroyed?
  destroyed = true
else
  destroyed = false
  reason = element.errors.collect { |_attr, msg| msg }.join(';')
end

if destroyed
  AuditEvent.success(audit)
  add_flash(_("%{model} \"%{name}\": Delete successful") % {:model => model_name, :name => record_name})
else
  add_flash(_("%{model} \"%{name}\": Error during delete: %{error_msg}") %
    {:model => model_name, :name => record_name, :error_msg => reason}, :error)
end
```